src/tools/w3dviewer/emitterinstancelist.cpp

```diff
@@ -0,0 +1,224 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief W3D View emitter instance list
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "emitterinstancelist.h"
+
+EmitterInstanceList::~EmitterInstanceList()
+{
+    Clear_Emitters();
+}
+
+void EmitterInstanceList::Set_Velocity(const Vector3 &value)
+{
+    ParticleEmitterDefClass::Set_Velocity(value);
+
+    for (int i = 0; i < m_emitters.Count(); i++) {
+        m_emitters[i]->Set_Base_Velocity(value);
+    }
+}
+
+void EmitterInstanceList::Set_Acceleration(const Vector3 &value)
+{
+    ParticleEmitterDefClass::Set_Acceleration(value);
+
+    for (int i = 0; i < m_emitters.Count(); i++) {
+        m_emitters[i]->Set_Acceleration(value);
+    }
+}
+
+void EmitterInstanceList::Set_Burst_Size(unsigned int size)
+{
+    ParticleEmitterDefClass::Set_Burst_Size(size);
+
+    for (int i = 0; i < m_emitters.Count(); i++) {
+        m_emitters[i]->Set_Burst_Size(size);
+    }
+}
+
+void EmitterInstanceList::Set_Outward_Vel(float value)
+{
+    ParticleEmitterDefClass::Set_Outward_Vel(value);
+
+    for (int i = 0; i < m_emitters.Count(); i++) {
+        m_emitters[i]->Set_Outwards_Velocity(value);
+    }
+}
+
+void EmitterInstanceList::Set_Vel_Inherit(float value)
+{
+    ParticleEmitterDefClass::Set_Vel_Inherit(value);
+
+    for (int i = 0; i < m_emitters.Count(); i++) {
+        m_emitters[i]->Set_Velocity_Inheritance_Factor(value);
+    }
+}
+
+void EmitterInstanceList::Set_Velocity_Random(Vector3Randomizer *randomizer)
+{
+    ParticleEmitterDefClass::Set_Velocity_Random(randomizer);
+
+    if (randomizer != nullptr) {
+        for (int i = 0; i < m_emitters.Count(); i++) {
+            m_emitters[i]->Set_Velocity_Randomizer(randomizer->Clone());
+        }
+    }
+}
+
+void EmitterInstanceList::Get_Color_Keyframes(ParticlePropertyStruct<Vector3> &keyframes) const
+{
+    ParticleEmitterDefClass::Get_Color_Keyframes(keyframes);
+
+    for (unsigned int i = 0; i < keyframes.NumKeyFrames; i++) {
+        if (keyframes.Values[i].X < 0.000001f) {

```

Comment:
You can make this epsilon a named constexpr in this file and then use it everywhere:
```
namespace
{
constexpr float s_keyFrameEpsilon = 0.000001f;
}
```

Reply:
Fixed.